api/configuration/v1alpha1/credential_basic_auth_types.go

```diff
@@ -0,0 +1,159 @@
+	Status CredentialBasicAuthStatus `json:"status,omitempty"`
+}
+
+func (s *CredentialBasicAuth) initKonnectStatus() {
+	s.Status.Konnect = &konnectv1alpha1.KonnectEntityStatusWithControlPlaneRef{}
+}
+
+// GetKonnectStatus returns the Konnect status contained in the CredentialBasicAuth status.
+func (s *CredentialBasicAuth) GetKonnectStatus() *konnectv1alpha1.KonnectEntityStatus {
+	if s.Status.Konnect == nil {
+		return nil
+	}
+	return &s.Status.Konnect.KonnectEntityStatus
+}
+
+// GetKonnectID returns the Konnect ID in the CredentialBasicAuth status.
+func (s *CredentialBasicAuth) GetKonnectID() string {
+	if s.Status.Konnect == nil {
+		return ""
+	}
+	return s.Status.Konnect.ID
+}
+
+// SetKonnectID sets the Konnect ID in the CredentialBasicAuth status.
+func (s *CredentialBasicAuth) SetKonnectID(id string) {
+	if s.Status.Konnect == nil {
+		s.initKonnectStatus()
+	}
+	s.Status.Konnect.ID = id
+}
+
+// GetControlPlaneID returns the ControlPlane ID in the CredentialBasicAuth status.
+func (s *CredentialBasicAuth) GetControlPlaneID() string {
+	if s.Status.Konnect == nil {
+		return ""
+	}
+	return s.Status.Konnect.ControlPlaneID
+}
+
+// SetControlPlaneID sets the ControlPlane ID in the CredentialBasicAuth status.
+func (s *CredentialBasicAuth) SetControlPlaneID(id string) {
+	if s.Status.Konnect == nil {
+		s.initKonnectStatus()
+	}
+	s.Status.Konnect.ControlPlaneID = id
+}
+
+// GetTypeName returns the CredentialBasicAuth Kind name
+func (s CredentialBasicAuth) GetTypeName() string {
+	return "CredentialBasicAuth"
+}
+
+// GetConditions returns the Status Conditions
+func (s *CredentialBasicAuth) GetConditions() []metav1.Condition {
+	return s.Status.Conditions
+}
+
+// SetConditions sets the Status Conditions
+func (s *CredentialBasicAuth) SetConditions(conditions []metav1.Condition) {
+	s.Status.Conditions = conditions
+}
+
+// CredentialBasicAuthSpec defines specification of a Kong Route.
+type CredentialBasicAuthSpec struct {
+	// ConsumerRef is a reference to a Consumer this CredentialBasicAuth is associated with.
+	//
+	// +kubebuilder:validation:Required
+	ConsumerRef corev1.LocalObjectReference `json:"consumerRef"`
+
+	// SecretRef is a reference to a Secret this CredentialBasicAuth is associated with.
+	//
+	// +kubebuilder:validation:Required
+	SecretRef corev1.LocalObjectReference `json:"secretRef"`
+
+	CredentialBasicAuthAPISpec `json:",inline"`
+}
+
+// CredentialBasicAuthAPISpec defines specification of a BasicAuth credential.
+type CredentialBasicAuthAPISpec struct {
+	// Password is the password for the BasicAuth credential.
+	Password *string `json:"password,omitempty"`
+	// Tags is a list of tags for the BasicAuth credential.
+	Tags []string `json:"tags,omitempty"`
+	// Username is the username for the BasicAuth credential.
+	Username *string `json:"username,omitempty"`
```

Comment:
Given that we know this field is not optional, why not use just `string`? Same applies to the password.

Reply:
And also we can add a `MinLength=1` constraint to make it non empty.

Reply:
Hm, so the API https://kong-platform-api.netlify.app/konnect/control-planes-config/v2/openapi.yaml/#tag/Basic-auth-credentials/operation/create-basic-auth-with-consumer does not seem to require either of those but Konnect UI does require both:

<img width="685" alt="image" src="https://github.com/user-attachments/assets/be851f28-1b84-495c-91a5-cc2e2c5d81ef">

I'll make the changes for this.

Reply:
46384cea11836e799f2dcc7d942f4dc620ebd396